driver-core/src/main/java/com/datastax/driver/core/ParseUtils.java

```diff
@@ -428,45 +460,114 @@ public static String formatTime(long value) {
+            return "\"\"";
+        else
+            return "''";
+    }
+
+
+    /**
+     * Quotes text and escapes any existing quotes in the text.
+     * {@code String.replace()} is a bit too inefficient (see JAVA-67, JAVA-1262).
      *
-     * @param text        The text.
-     * @param search      The character to search for.
-     * @param replacement The replacement.
-     * @return The text with all occurrences of {@code search} replaced with {@code replacement}.
+     * @param text      The text.
+     * @param quoteChar The character to use as a quote.
+     * @return The text with surrounded in quotes with all existing quotes escaped with (i.e. ' becomes '')
      */
-    private static String replaceChar(String text, char search, String replacement) {
+    private static String quote(String text, char quoteChar) {
         if (text == null || text.isEmpty())
-            return "";
+            return emptyQuoted(quoteChar);
 
         int nbMatch = 0;
         int start = -1;
         do {
-            start = text.indexOf(search, start + 1);
+            start = text.indexOf(quoteChar, start + 1);
             if (start != -1)
                 ++nbMatch;
         } while (start != -1);
 
-        if (nbMatch == 0)
+        // 2 for beginning and end quotes.
+        // length for original text
+        // nbMatch for escape characters to add to quotes to be escaped.
+        int newLength = 2 + text.length() + nbMatch;
+        char[] result = new char[newLength];
+        result[0] = quoteChar;
+        result[newLength - 1] = quoteChar;
+        int newIdx = 1;
+        for (int i = 0; i < text.length(); i++) {
+            char c = text.charAt(i);
+            if (c == quoteChar) {
+                // escape quote with another occurrence.
+                result[newIdx++] = c;
+                result[newIdx++] = c;
+            } else {
+                result[newIdx++] = c;
+            }
+        }
+        return new String(result);
+    }
+
+    /**
+     * Unquotes text and unescapes non surrounding quotes.
+     * {@code String.replace()} is a bit too inefficient (see JAVA-67, JAVA-1262).
+     *
+     * @param text      The text
+     * @param quoteChar The character to use as a quote.
+     * @return The text with surrounding quotes removed and non surrounding quotes unescaped (i.e. '' becomes ')
+     */
+    private static String unquote(String text, char quoteChar) {
+        if (!isQuoted(text, quoteChar))
             return text;
 
-        int newLength = text.length() + nbMatch * (replacement.length() - 1);
+        String search = emptyQuoted(quoteChar);
+        int nbMatch = 0;
+        int start = -2;
+        do {
+            start = text.indexOf(search, start + 2);
+            // ignore the second to last character occurrence, as the last character is a quote.
+            if (start != -1 && start != text.length() - 2)
+                ++nbMatch;
+        } while (start != -1);
+
```

Comment:
Hrmm, I think i'm coming around on that.  The empty strings case is already covered for both quote/unquote.  I'll take care of simple identifiers.

The only problem is the quoting case.   I had done some testing and:

```
"'" + text + "'";
```

Seems to be optimized by the compiler (in terms of memory allocation) because the quotes are constant, but ```quoteChar + text + quoteChar``` isn't because quoteChar isn't a constant.   I'll do some more testing around that.